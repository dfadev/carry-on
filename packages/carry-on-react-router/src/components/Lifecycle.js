import React from "react";

class Lifecycle extends React.Component {
  componentDidMount() {
    const { onMount } = this.props;
    if (onMount) onMount.call(this, this);
  }

  componentDidUpdate(prevProps) {
    const { onUpdate } = this.props;
    if (onUpdate) onUpdate.call(this, this, prevProps);
  }

  componentWillUnmount() {
    const { onUnmount } = this.props;
    if (onUnmount) onUnmount.call(this, this);
  }

  render() {
    return null;
  }
}

export default Lifecycle;
