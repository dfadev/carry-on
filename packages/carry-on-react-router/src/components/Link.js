import React from "react";
import { carryOn } from "carry-on-react";
import { getIn } from "carry-on-utils";

const Link = carryOn(
  { id: "Link" },
  (
    {
      history = "app.history",
      innerRef,
      replace,
      to,
      onClick,
      target,
      force,
      ...rest
    },
    state
  ) => {
    const { handleClick, getHref } = getIn(state, history);

    return (
      <a
        {...rest}
        onClick={handleClick({ replace, to, onClick, target, force })}
        href={getHref(to)}
        ref={innerRef}
      />
    );
  }
);

export default Link;
