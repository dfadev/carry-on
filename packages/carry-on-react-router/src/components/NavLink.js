import React from "react";
import { carryOn } from "carry-on-react";
import { getIn } from "carry-on-utils";

function joinClassnames(...classnames) {
  return classnames.filter(i => i).join(" ");
}

const escapedRE = /([.+*?=^!:${}()[\]|/\\])/g;

const NavLink = carryOn(
  { id: "NavLink" },
  (
    {
      history = "app.history",
      "aria-current": ariaCurrent = "page",
      activeClassName = "active",
      activeStyle,
      className: classNameProp,
      exact,
      isActive: isActiveProp,
      location: loc,
      style: styleProp,
      strict,
      sensitive,
      to,
      replace,
      onClick,
      target,
      innerRef,
      ...rest
    },
    state
  ) => {
    const hist = getIn(state, history);
    const location = loc || hist.location;
    const path = typeof to === "object" ? to.pathname : to;
    const escapedPath = path && path.replace(escapedRE, "\\$1");
    const match = hist.matchPath({
      path: escapedPath,
      exact,
      strict,
      sensitive
    });
    const isActive = !!(isActiveProp ? isActiveProp(match, location) : match);
    const className = isActive
      ? joinClassnames(classNameProp, activeClassName)
      : classNameProp;
    const style = isActive ? { ...styleProp, ...activeStyle } : styleProp;
    const { handleClick, getHref } = hist;

    /* eslint-disable jsx-a11y/anchor-has-content */
    return (
      <a
        {...rest}
        onClick={handleClick({ replace, to, onClick, target })}
        href={getHref(to)}
        ref={innerRef}
        style={style}
        className={className}
        aria-current={(isActive && ariaCurrent) || null}
      />
    );
  }
);

export default NavLink;
