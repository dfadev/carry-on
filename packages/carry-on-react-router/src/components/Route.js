import React from "react";
import { carryOn } from "carry-on-react";
import { getIn } from "carry-on-utils";

function isEmptyChildren(children) {
  return React.Children.count(children) === 0;
}

const Route = carryOn((props, state) => {
  const {
    history = "app.history",
    location: loc,
    computedMatch,
    path,
    children: childs,
    component,
    render
  } = props;

  const hist = getIn(state, history);
  const location = loc || hist.location;

  let match;
  if (computedMatch) match = computedMatch;
  else if (path) match = hist.matchPath(props);
  //else match = hist.match; // ???

  const renderProps = { location, match };

  let children = childs;
  if (Array.isArray(children) && children.length === 0)
    children = null;

  if (typeof children === "function") {
    children = children(renderProps);
    if (children === undefined) children = null;
  }

  if (children && !isEmptyChildren(children)) return children;
  if (match)
    if (component) return React.createElement(component, renderProps);
    else if (render) return render(renderProps);

  return null;
});

Route.displayName = "Route";

export default Route;
