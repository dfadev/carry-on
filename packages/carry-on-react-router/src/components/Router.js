/* eslint-disable max-classes-per-file */
import React, { useMemo } from "react";
import {
  withStore,
  State,
  Register,
  Middleware,
  OnUnmount,
  Render
} from "carry-on-react";
import {
  createBrowserHistory,
  createHashHistory,
  createMemoryHistory
} from "history";
import createStaticHistory from "../createStaticHistory";
import router from "../router";

function GenericRouter(createHistory = createBrowserHistory) {
  return withStore(({ children = null, path = "app.history", ...props }) => {
    const rtr = useMemo(() => router(createHistory(props), path), [path]);

    return (
      <State {...props}>
        <Register>{rtr.state}</Register>
        <Middleware>{rtr.middleware}</Middleware>
        <OnUnmount>
          {state =>
            state &&
            state.app &&
            state.app.history &&
            state.app.history.unlisten &&
            state.app.history.unlisten()
          }
        </OnUnmount>
        <Render>{() => children || null}</Render>
      </State>
    );
  });
}

export const Router = GenericRouter(props => props.history);
export const MemoryRouter = GenericRouter(createMemoryHistory);
export const BrowserRouter = GenericRouter(createBrowserHistory);
export const HashRouter = GenericRouter(createHashHistory);
export const StaticRouter = GenericRouter(createStaticHistory);
