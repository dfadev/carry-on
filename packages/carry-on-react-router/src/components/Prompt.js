import React from "react";
import { carryOn } from "carry-on-react";
import { getIn } from "carry-on-utils";
import Lifecycle from "./Lifecycle";

/**
 * The public API for prompting the user before navigating away from a screen.
 */
const Prompt = carryOn(
  { id: "Prompt" },
  ({ history = "app.history", message, when = true }, state) => {
    const hist = getIn(state, history);

    if (!when || hist.staticContext) return null;

    const method = hist.block;

    return (
      <Lifecycle
        onMount={self => {
          self.release = method(message);
        }}
        onUpdate={(self, prevProps) => {
          if (prevProps.message !== message) {
            self.release();
            self.release = method(message);
          }
        }}
        onUnmount={self => {
          self.release();
        }}
        message={message}
      />
    );
  }
);

export default Prompt;
