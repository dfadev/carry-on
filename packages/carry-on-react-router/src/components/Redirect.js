import React from "react";
import { createLocation, locationsAreEqual } from "history";
import { carryOn } from "carry-on-react";
import { getIn } from "carry-on-utils";

import Lifecycle from "./Lifecycle";
import generatePath from "../generatePath";

const Redirect = carryOn(
  { id: "Redirect" },
  ({ history = "app.history", computedMatch, to, push = false }, state) => {
    const hist = getIn(state, history);
    const method = push ? hist.push : hist.replace;

    /* eslint-disable no-nested-ternary */
    const location = createLocation(
      computedMatch
        ? typeof to === "string"
          ? generatePath(to, computedMatch.params)
          : {
              ...to,
              pathname: generatePath(to.pathname, computedMatch.params)
            }
        : to
    );

    // When rendering in a static context, set the new location immediately.
    if (hist.staticContext) {
      method(location);
      return null;
    }

    return (
      <Lifecycle
        onMount={() => {
          method(location);
        }}
        onUpdate={(_, prevProps) => {
          if (!locationsAreEqual(prevProps.to, location)) {
            method(location);
          }
        }}
        to={to}
      />
    );
  }
);

export default Redirect;
