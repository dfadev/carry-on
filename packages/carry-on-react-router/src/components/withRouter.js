/* eslint-disable react/jsx-props-no-spreading */
import React from "react";
import hoistStatics from "hoist-non-react-statics";
import Route from "./Route";

function withRouter(Component) {
  const C = props => {
    const { wrappedComponentRef, ...remainingProps } = props;

    return (
      <Route>
        {routeComponentProps => (
          <Component
            {...remainingProps}
            {...routeComponentProps}
            ref={wrappedComponentRef}
          />
        )}
      </Route>
    );
  };

  C.displayName = `withRouter(${Component.displayName || Component.name})`;
  C.WrappedComponent = Component;

  return hoistStatics(C, Component);
}

export default withRouter;
