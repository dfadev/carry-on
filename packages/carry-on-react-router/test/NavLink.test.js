import React from "react";
import { render } from "@testing-library/react";
import { initStores } from "carry-on-store";
import { MemoryRouter } from "../src/components/Router";
import NavLink from "../src/components/NavLink";
import withRouter from "../src/components/withRouter";

describe("A <NavLink>", () => {
  afterEach(() => {
    initStores();
  });

  describe("when active", () => {
    it("applies its default activeClassName", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink to="/pizza">Pizza!</NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("applies a custom activeClassName instead of the default", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink to="/pizza" activeClassName="selected">
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("applies its activeStyle", () => {
      const defaultStyle = { color: "black" };
      const activeStyle = { color: "red" };

      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink to="/pizza" style={defaultStyle} activeStyle={activeStyle}>
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("applies the default aria-current", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink to="/pizza">Pizza!</NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("applies a custom aria-current instead of the default", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink to="/pizza" aria-current="true">
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("handles locations without a pathname", () => {
      expect(() => {
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink to={{ search: "foo=bar" }}>Pizza!</NavLink>
          </MemoryRouter>
        ).asFragment();
      }).not.toThrow();
    });

    it("automatically escapes special characters in the path", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza (1)"]}>
            <NavLink to="/pizza (1)">Pizza!</NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("renders child components that use withRouter", () => {
      class WrappedComponent extends React.Component {
        render() {
          return null;
        }
      }

      const Component = withRouter(WrappedComponent);

      let ref;
      render(
        <MemoryRouter initialEntries={["/pizza"]}>
          <NavLink to="/pizza">
            <Component wrappedComponentRef={r => (ref = r)} />
          </NavLink>
        </MemoryRouter>
      );

      expect(ref instanceof WrappedComponent).toBe(true);
    });
  });

  describe("when inactive", () => {
    it("does not apply its default activeClassName", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink to="/salad">Salad?</NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("does not apply its activeClassName", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink to="/salad" activeClassName="selected">
              Salad?
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("does not apply its activeStyle", () => {
      const defaultStyle = { color: "black" };
      const activeStyle = { color: "red" };

      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink to="/salad" style={defaultStyle} activeStyle={activeStyle}>
              Salad?
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("does not apply an aria-current value if no override value is given", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink to="/salad" activeClassName="selected" aria-current="page">
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("does not apply an aria-current value if an override value is given", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink to="/salad" activeClassName="selected">
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("renders child components that use withRouter", () => {
      class WrappedComponent extends React.Component {
        render() {
          return null;
        }
      }

      const Component = withRouter(WrappedComponent);

      let ref;
      render(
        <MemoryRouter initialEntries={["/pizza"]}>
          <NavLink exact to="/salad">
            <Component wrappedComponentRef={r => (ref = r)} />
          </NavLink>
        </MemoryRouter>
      ).asFragment();

      expect(ref instanceof WrappedComponent).toBe(true);
    });
  });

  describe("isActive", () => {
    it("applies active default props when isActive returns true", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink to="/pizza" isActive={() => true}>
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("applies activeClassName when isActive returns true", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink
              to="/pizza"
              activeClassName="selected"
              isActive={() => true}
            >
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("does not apply default activeClassName when isActive returns false", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink to="/pizza" isActive={() => false}>
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("does not apply custom activeClassName when isActive returns false", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink
              to="/pizza"
              activeClassName="selected"
              isActive={() => false}
            >
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });
  });

  it("does not do exact matching by default", () => {
    expect(
      render(
        <MemoryRouter initialEntries={["/pizza/anchovies"]}>
          <NavLink to="/pizza" activeClassName="active">
            Pizza!
          </NavLink>
        </MemoryRouter>
      ).asFragment()
    ).toMatchSnapshot();
  });

  describe("with `exact=true`", () => {
    it("applies default activeClassName for exact matches", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink exact to="/pizza">
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("does not apply default activeClassName for partial matches", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza/anchovies"]}>
            <NavLink exact to="/pizza">
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("applies custom activeClassName for exact matches", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink exact to="/pizza" activeClassName="selected">
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("applies custom activeClassName for partial matches", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza/anchovies"]}>
            <NavLink exact to="/pizza" activeClassName="selected">
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });
  });

  // does not apply due to path-to-regexp change
  // it("does not do strict matching by default", () => {
  // expect(
  // render(
  // <MemoryRouter initialEntries={["/pizza"]}>
  // <NavLink to="/pizza/">Pizza!</NavLink>
  // </MemoryRouter>
  // ).asFragment()
  // ).toMatchSnapshot();
  // });

  describe("with `strict=true`", () => {
    it("applies default activeClassName for strict matches", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza/"]}>
            <NavLink strict to="/pizza/">
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("does not apply default activeClassName for non-strict matches", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink strict to="/pizza/">
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("applies custom activeClassName for strict matches", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza/"]}>
            <NavLink strict to="/pizza/" activeClassName="selected">
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("does not apply custom activeClassName for non-strict matches", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink strict to="/pizza/" activeClassName="selected">
              Pizza!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });
  });

  describe("the `location` prop", () => {
    it("overrides the current location", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pizza"]}>
            <NavLink to="/pasta" location={{ pathname: "/pasta" }}>
              Pasta!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });

    it("is not overwritten by the current location", () => {
      expect(
        render(
          <MemoryRouter initialEntries={["/pasta"]}>
            <NavLink to="/pasta" location={{ pathname: "/pizza" }}>
              Pasta!
            </NavLink>
          </MemoryRouter>
        ).asFragment()
      ).toMatchSnapshot();
    });
  });
});
