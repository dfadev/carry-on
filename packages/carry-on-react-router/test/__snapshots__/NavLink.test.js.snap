// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A <NavLink> does not do exact matching by default 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="active"
    href="/pizza"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> does not do strict matching by default 1`] = `
<DocumentFragment>
  <a
    href="/pizza/"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> isActive applies active default props when isActive returns true 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="active"
    href="/pizza"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> isActive applies activeClassName when isActive returns true 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="selected"
    href="/pizza"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> isActive does not apply custom activeClassName when isActive returns false 1`] = `
<DocumentFragment>
  <a
    href="/pizza"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> isActive does not apply default activeClassName when isActive returns false 1`] = `
<DocumentFragment>
  <a
    href="/pizza"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> the \`location\` prop is not overwritten by the current location 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="active"
    href="/pasta"
  >
    Pasta!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> the \`location\` prop overrides the current location 1`] = `
<DocumentFragment>
  <a
    href="/pasta"
  >
    Pasta!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> when active applies a custom activeClassName instead of the default 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="selected"
    href="/pizza"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> when active applies a custom aria-current instead of the default 1`] = `
<DocumentFragment>
  <a
    aria-current="true"
    class="active"
    href="/pizza"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> when active applies its activeStyle 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="active"
    href="/pizza"
    style="color: red;"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> when active applies its default activeClassName 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="active"
    href="/pizza"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> when active applies the default aria-current 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="active"
    href="/pizza"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> when active it automatically escapes special characters in the path 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="active"
    href="/pizza (1)"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> when inactive does not apply an aria-current value if an override value is given 1`] = `
<DocumentFragment>
  <a
    href="/salad"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> when inactive does not apply an aria-current value if no override value is given 1`] = `
<DocumentFragment>
  <a
    href="/salad"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> when inactive does not apply its activeClassName 1`] = `
<DocumentFragment>
  <a
    href="/salad"
  >
    Salad?
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> when inactive does not apply its activeStyle 1`] = `
<DocumentFragment>
  <a
    href="/salad"
    style="color: black;"
  >
    Salad?
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> when inactive does not apply its default activeClassName 1`] = `
<DocumentFragment>
  <a
    href="/salad"
  >
    Salad?
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> with \`exact=true\` applies custom activeClassName for exact matches 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="selected"
    href="/pizza"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> with \`exact=true\` applies custom activeClassName for partial matches 1`] = `
<DocumentFragment>
  <a
    href="/pizza"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> with \`exact=true\` applies default activeClassName for exact matches 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="active"
    href="/pizza"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> with \`exact=true\` does not apply default activeClassName for partial matches 1`] = `
<DocumentFragment>
  <a
    href="/pizza"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> with \`strict=true\` applies custom activeClassName for strict matches 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="selected"
    href="/pizza/"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> with \`strict=true\` applies default activeClassName for strict matches 1`] = `
<DocumentFragment>
  <a
    aria-current="page"
    class="active"
    href="/pizza/"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> with \`strict=true\` does not apply custom activeClassName for non-strict matches 1`] = `
<DocumentFragment>
  <a
    href="/pizza/"
  >
    Pizza!
  </a>
</DocumentFragment>
`;

exports[`A <NavLink> with \`strict=true\` does not apply default activeClassName for non-strict matches 1`] = `
<DocumentFragment>
  <a
    href="/pizza/"
  >
    Pizza!
  </a>
</DocumentFragment>
`;
